### PR TITLE
Style changes in `RecoTracker/DeDx`

### PR DESCRIPTION
**Pull request by tvami**
#### PR description:

I was going to do some development in `RecoTracker/DeDx` but noticed that many thing dont comply with the coding rules
https://cms-sw.github.io/cms_coding_rules.html

So this PR does several style changes in that package regarding the following rules:
* 2.7 For namespaces use lower case
* 2.8 Start method names with lowercase, use upper case initials for following words
* 2.9 Start data member names with lower case. A trailing “_” is the preferred method to distinguish a data member from the getter method
* 6.3 Only header files that expose a public interface should go into the Package/Subpackage/interface directory.

Besides this, 
* some variable names were changed (more descriptive name), 
* French comments were translated to English (if that's incorrect let me know...)
* Spelling mistakes were fixed (`Reccord`)
* Tried to reduce code duplication
* Removed commented out code

Changes are just cosmetics, so no changes in the tests are expected.

After this merges, I have a follow-up with some real changes included.

#### PR validation:

` scram b runtests` runs fine (except for the case when I have file opening errors)

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A

### CODE REVIEW

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by Martin-Grunewald** [comment]
Indeed, please code a customisation function in ```HLTrigger/Configuration/python/customizeHLTforCMSSW.py``` to make the change if the old parameter ```Reccord``` occurs in the py config. See the file for examples - and make sure you add a call to the customisation function at the end of that file.

**Comment by tvami** [comment]
@cmsbuild , please test

File: HLTrigger/Configuration/python/customizeHLTforCMSSW.py [hunk 1]
```diff
@@ -156,3 +155,16 @@ def customizeHLTforCMSSW(process, menuType="GRun"):
...
+
+    for prod in producers_by_type(process, 'DeDxEstimatorProducer'):
+        if hasattr(prod, 'Reccord'):
+            delattr(prod, 'Reccord')
+            setattr(prod, 'Record', "SiStripDeDxMip_3D_Rcd")
+
+    process = customiseFor37231(process)
+
+    return process
+
```
Comment (Martin-Grunewald) [inline HLTrigger/Configuration/python/customizeHLTforCMSSW.py:170]: You need to move this function up, and move the line:
```
process = customiseFor37231(process)
```
into the function ```def customizeHLTforCMSSW```:
```

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by francescobrivio** [comment]
+alca

 - tests clean
 - style changes only in the `alca` concerned part

File: RecoTracker/DeDx/src/DeDxTools.cc [hunk 2]
```diff
@@ -280,23 +280,23 @@ namespace DeDxTools {
...
-    if (Reccord == "SiStripDeDxProton_3D_Rcd") {
+    if (Record == "SiStripDeDxProton_3D_Rcd") {
       return iCC.esConsumes<H3DD, SiStripDeDxProton_3D_Rcd, edm::Transition::BeginRun>();
     }
-    if (Reccord == "SiStripDeDxElectron_3D_Rcd") {
+    if (Record == "SiStripDeDxElectron_3D_Rcd") {
       return iCC.esConsumes<H3DD, SiStripDeDxElectron_3D_Rcd, edm::Transition::BeginRun>();
     }
-    throw cms::Exception("WrongReccord for dEdx") << "The reccord : " << Reccord << "is unknown\n";
+    throw cms::Exception("WrongRecord for dEdx") << "The reccord : " << Record << "is unknown\n";
```
Comment (mmusich) [inline RecoTracker/DeDx/src/DeDxTools.cc:299]:
```suggestion
    throw cms::Exception("WrongRecord for dEdx") << "The record : " << Record << "is unknown\n";
```

File: HLTrigger/Configuration/python/customizeHLTforCMSSW.py [hunk 3]
```diff
@@ -144,6 +144,15 @@ def customiseFor2018Input(process):
 
     return process
 
+def customiseFor37231(process):
+    """ Customisation to fix the typo of Reccord in PR 37231 (https://github.com/cms-sw/cmssw/pull/37231) """
+
+    for prod in producers_by_type(process, 'DeDxEstimatorProducer'):
+        if hasattr(prod, 'Reccord'):
+            delattr(prod, 'Reccord')
+            setattr(prod, 'Record', cms.string("SiStripDeDxMip_3D_Rcd"))
```
[the remark this answers is not in the document]
Comment (tvami) [reply HLTrigger/Configuration/python/customizeHLTforCMSSW.py:153]: I cant. That's what I tried first, it fails by saying that this is a new variable and it's type is unknown.

[the remark this answers is not in the document]
Comment (missirol) [reply HLTrigger/Configuration/python/customizeHLTforCMSSW.py:153]: I tested the suggestion, and all the `addOnTests` passed.
```
diff --git a/HLTrigger/Configuration/python/customizeHLTforCMSSW.py b/HLTrigger/Configuration/python/customizeHLTforCMSSW.py
index eec54d17696..d471f7d3794 100644
--- a/HLTrigger/Configuration/python/customizeHLTforCMSSW.py
+++ b/HLTrigger/Configuration/python/customizeHLTforCMSSW.py
@@ -149,8 +149,8 @@ def customiseFor37231(process):
 
     for prod in producers_by_type(process, 'DeDxEstimatorProducer'):
         if hasattr(prod, 'Reccord'):
+            prod.Record = prod.Reccord
             delattr(prod, 'Reccord')
-            setattr(prod, 'Record', cms.string("SiStripDeDxMip_3D_Rcd"))
 
     return process
```

[the remark this answers is not in the document]
Comment (tvami) [reply HLTrigger/Configuration/python/customizeHLTforCMSSW.py:153]: Ahh sorry, I responded from my phone w/o carefully looking at what you wrote. What I tested earlier was `setattr(prod, 'Record', "SiStripDeDxMip_3D_Rcd")` that didnt work. I'll commit your suggestion in a second.

**Comment by tvami** [comment]
I will squash the commits, but let's see first if @cms-sw/reconstruction-l2 has any comments or not

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
+alca
 * see https://github.com/cms-sw/cmssw/pull/37231#issuecomment-1068391094

**Comment by clacaputo** [comment]
+reconstruction

- style changes in `RecoTracker/DeDx` in compliance with cms coding rules
- no reco changes expected or observed

**Comment by Martin-Grunewald** [comment]
+1

**Comment by tvami** [comment]
Hi @sbein the fast-sim changes are trivial, can you please sign it?

**Comment by qliphy** [comment]
merge
As requested there is a follow-up PR waiting for this one.   @sbein  let's know if you have any concern.

**Comment by qliphy** [comment]
merge

**Pull request merged**